FV3/atmos_cubed_sphere/model/fv_update_phys.F90

```diff
@@ -569,22 +569,49 @@ subroutine fv_update_phys ( dt, is, ie, js, je, isd, ied, jsd, jed, ng, nq,
                             zvir, ptop, ak, bk, ts, ps, delp, ua, va, pt,    &
                             nwat, q,  phis, gridstruct, bd, domain )
 
-         if (allocated(nudge_diag%nudge_t_dt)) nudge_diag%nudge_t_dt = (pt(is:ie,js:je,:) - nudge_diag%nudge_t_dt) / dt
-         if (allocated(nudge_diag%nudge_ps_dt)) nudge_diag%nudge_ps_dt = (ps(is:ie,js:je) - nudge_diag%nudge_ps_dt) / dt
-         if (allocated(nudge_diag%nudge_delp_dt)) nudge_diag%nudge_delp_dt = (delp(is:ie,js:je,:) - nudge_diag%nudge_delp_dt) / dt
-         qv_dt_nudge = (q(is:ie,js:je,:,sphum) - qv_dt_nudge) / dt
-         if (allocated(nudge_diag%nudge_q_dt)) nudge_diag%nudge_q_dt = qv_dt_nudge
-
-          ! Note that u_dt and v_dt are already tendencies, so we do not need to divide by dt.
-         if (allocated(nudge_diag%nudge_u_dt)) nudge_diag%nudge_u_dt = (u_dt(is:ie,js:je,:) - nudge_diag%nudge_u_dt)
-         if (allocated(nudge_diag%nudge_v_dt)) nudge_diag%nudge_v_dt = (v_dt(is:ie,js:je,:) - nudge_diag%nudge_v_dt)
-
-         call compute_column_moistening_implied_by_nudging(qv_dt_nudge(is:ie,js:je,1:npz), &
-           delp(is:ie,js:je,1:npz), &
-           is, ie, js, je, npz, &
-           column_moistening_implied_by_nudging(is:ie,js:je))
-
-         if (allocated(nudge_diag%column_moistening)) nudge_diag%column_moistening = column_moistening_implied_by_nudging(is:ie,js:je)
+        if (allocated(nudge_diag%nudge_t_dt)) then
+          nudge_diag%nudge_t_dt = (pt(is:ie,js:je,:) - nudge_diag%nudge_t_dt) / dt
+          if (allocated(nudge_diag%column_heating)) then
+            call compute_column_integral(nudge_diag%nudge_t_dt, &
+              delp(is:ie,js:je,1:npz), &
+              is, ie, js, je, npz, &
+              nudge_diag%column_heating(is:ie,js:je))
+            nudge_diag%column_heating(is:ie,js:je) = con_cp * nudge_diag%column_heating(is:ie,js:je)
```

Comment:
Should the scale factor be dependent on whether we are running in hydrostatic or non-hydrostatic mode (i.e. Cp for hydrostatic and Cv for non-hydrostatic)?

Reply:
I don't think so, given that the temperature tendency is computed and applied to the state in the same way for both hydrostatic and non-hydrostatic mode, e.g. see

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/1fe12b78ce57668300d9d327ed07b51f7f3bc8d8/FV3/atmos_cubed_sphere/tools/fv_nudge.F90#L637

(note the observational temperature is getting converted from virtual temperature) and

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/1fe12b78ce57668300d9d327ed07b51f7f3bc8d8/FV3/atmos_cubed_sphere/tools/fv_nudge.F90#L703

Using Cp is also consistent with how we compute column-integrated heating of tendencies in other places (e.g. ML tendencies).